datasafe-rest/src/main/java/de/adorsys/datasafe/rest/controller/InboxController.java

```diff
@@ -0,0 +1,74 @@
+package de.adorsys.datasafe.rest.controller;
+
+import de.adorsys.datasafe.business.api.types.UserID;
+import de.adorsys.datasafe.business.api.types.UserIDAuth;
+import de.adorsys.datasafe.business.api.types.actions.ReadRequest;
+import de.adorsys.datasafe.business.api.types.actions.RemoveRequest;
+import de.adorsys.datasafe.business.api.types.actions.WriteRequest;
+import de.adorsys.datasafe.business.api.types.keystore.ReadKeyPassword;
+import de.adorsys.datasafe.business.api.types.resource.BasePrivateResource;
+import de.adorsys.datasafe.business.api.types.resource.PrivateResource;
+import de.adorsys.datasafe.business.impl.service.DefaultDatasafeServices;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.util.StreamUtils;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+
+@Slf4j
+@RestController
+@RequestMapping("/inbox")
+public class InboxController {
+
+    private final DefaultDatasafeServices dataSafeService;
+
+    public InboxController(DefaultDatasafeServices dataSafeService) {
+        this.dataSafeService = dataSafeService;
+    }
+
+    @SneakyThrows
+    @PutMapping("/{path:.*}")
+    public void sendDocumentToInbox(@RequestHeader String user,
+                                    @PathVariable String path,
+                                    InputStream inputStream) {
+        UserID toUser = new UserID(user);
+        OutputStream stream = dataSafeService.inboxService().write(WriteRequest.forDefaultPublic(toUser, path));
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
```

Comment:
```
try (OutputStream os = dataSafeService.inboxService().write(WriteRequest.forDefaultPublic(toUser, path))) {
            StreamUtils.copy(inputStream, os);
        } finally {
            inputStream.close();
        }
```
Instead of all those unsafe operations

Reply:
ok